server/neptune/sync/crons/ratelimit_stats.go

```diff
@@ -0,0 +1,45 @@
+package crons
+
+import (
+	"context"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/metrics"
+	"github.com/uber-go/tally/v4"
+	"net/http"
+)
+
+const (
+	subScope  = "github.ratelimit"
+	remaining = "remaining"
+)
+
+type RateLimitStatCollector struct {
+	Scope          tally.Scope
+	ClientCreator  githubapp.ClientCreator
+	InstallationID int64
+}
+
+func NewRateLimitStats(scope tally.Scope, clientCreator githubapp.ClientCreator, installationID int64) *RateLimitStatCollector {
+	return &RateLimitStatCollector{
+		Scope:          scope.SubScope(subScope),
+		ClientCreator:  clientCreator,
+		InstallationID: installationID,
+	}
+}
+
+func (r *RateLimitStatCollector) Run(ctx context.Context) error {
+	installationClient, err := r.ClientCreator.NewInstallationClient(r.InstallationID)
+	if err != nil {
+		return errors.Wrap(err, "creating installation client")
+	}
```

Comment:
is this creating a new client every time? or does it pass back one thats already been created?

Reply:
ClientCreator comment:
```
// NewCachingClientCreator returns a ClientCreator that creates a GitHub client for installations of the app specified
// by the provided arguments. It uses an LRU cache of the provided capacity to store clients created for installations
// and returns cached clients when a cache hit exists.
```

See code itself for cache layer: https://github.com/palantir/go-githubapp/blob/develop/githubapp/caching_client_creator.go#L74